src/riak_kv.proto

```diff
@@ -291,3 +291,64 @@ message RpbBucketKeyPreflistItem {
     required bytes node = 2;
     required bool  primary = 3;
 }
+
+// Time Series
+
+// Dispatch a query to Riak
+message TsQueryReq {
+  // left optional to support parameterized queries in the future
+  optional TsInterpolation query = 1;
+}
+
+message TsQueryResp {
+  repeated TsColumnDescription columns = 1;
+  repeated TsRow rows = 2;
+}
+
+message TsPutReq {
+  required bytes table = 1;
+
+  // optional: omitting it should use table order
+  repeated TsColumnDescription columns = 2;
+
+  repeated TsRow rows = 3;
+}
+
+message TsPutResp {
+
+}
+
+message TsInterpolation {
+  required bytes base = 1;
+  repeated RpbPair interpolations = 2;
+}
+
+enum TsColumnType {
+  BINARY = 0;
+  INTEGER = 1;
+  NUMERIC = 2;
+  TIMESTAMP = 3;
+  BOOLEAN = 4;
+  SET = 5;
+  MAP = 6;
+}
+
+message TsColumnDescription {
+  required bytes name = 1;
+  required TsColumnType type = 2;
+  repeated TsColumnType complex_type = 3;
```

Comment:
Just curious as to what a `complex_type` might be and why it would be additional to the `type`